lib/stm32/f4/rcc.c

```diff
@@ -446,60 +458,78 @@ void rcc_osc_bypass_disable(enum rcc_osc osc)
 	}
 }
 
+/*
+ */
+void rcc_set_sysclk(enum rcc_osc clk) {
+	uint32_t clk_bits = RCC_CFGR_SW_HSI;
+	switch (clk) {
+		case HSI:
+			clk_bits = RCC_CFGR_SW_HSI;
+			break;
+		case HSE:
+			clk_bits = RCC_CFGR_SW_HSE;
+			break;
+		case PLL:
+			clk_bits = RCC_CFGR_SW_PLL;
+			break;
+		default:
+			clk_bits = RCC_CFGR_SW_HSI;
+			break;
+	}
+	RCC_CFGR = (RCC_CFGR & (~RCC_CFGR_SW_MASK)) |
+			   (clk_bits & RCC_CFGR_SW_MASK);
+	/* wait for the switch */
+	while (((RCC_CFGR >> RCC_CFGR_SWS_SHIFT) & RCC_CFGR_SW_MASK) != clk_bits) ;
+}
 
+enum rcc_osc rcc_get_sysclk(void) {
+	uint32_t clk_bits = (RCC_CFGR >> RCC_CFGR_SWS_SHIFT) & RCC_CFGR_SW_MASK;
+	switch (clk_bits) {
+		default:
+			return HSI;
```

Comment:
it was combined with the case 0: case so it ended up on top. Moved to the bottom.